### ISSUE TITLE
Support for libmpv 1.24.0

### ISSUE THREAD

**Issue by HarukaMa** [lhc70000/iina#34]
I couldn't use brew to install libmpv due to conflicts (I use self-compiled ffmpeg and x264 etc) so I tried to build mpv from git tree. I noticed the libmpv version is 1.24.0, so I changed all referencing in xcodeproj to `libmpv.1.24.0.dylib`. 

After successful compiling IINA without code signing (should make no differences?), I got the following error on launching IINA:

```
Fatal error: MPV API error: "invalid parameter", Return value: -4. 
The application will exit now.
```

Is it caused by API incompatible?

**Comment by lhc70000** [comment]
Since you are building by yourself, you may figure out in which line this error occurs by adding break points. It should be an API incompatibility. 

Updating libmpv version needs:

- newer C header files in `libmpv/include/mpv/`
- newer `MPVOption.swift`, etc. generated by `other/parse_doc.rb`
- other depended dylibs generated by `other/change_lib_dependencies.rb`   
- probably, change in IINA source code.

I'll consider updating libmpv in alpha 3 or 4. If you just want to build IINA, the dylibs provided in `libmpv/lib/` should be sufficient.

**Comment by HarukaMa** [comment]
After debugging, I have found the program failed in mpv: 

```c
(in main.c, function mp_initialize)
    if (!mpctx->playlist->first && !opts->player_idle_mode)
        return -3;
```

The playlist is empty and idle mode is false, so mpv init failed with return code -3, thus IINA thinks it has failed to initialize mpv.

Manually setting the idle mode when initializing would make the program survive the launch process:

```swift
    chkErr(mpv_set_option_string(mpv, MPVOption.ProgramBehavior.idle, "once"))
```

However when I stop playing a video, mpv will crash in some video output parts:

```
[0m2017-01-06 02:05:18.356581 IINA[71124:729898] MPV event: shutdown
Assertion failed: (!ctx->initialized), function free_ctx, file ../video/out/vo_opengl_cb.c, line 113.
```

--------------------

I just read the IINA project README again when I write this comment and seems I somehow missed the `If you are unwilling to use the provided dylibs` sentence. Thus I tried to rebuild IINA using provided 1.23.0 dylib, and everything works fine. I'm now wondering what differences in mpv building process would contribute to this huge behavior differences. 

I could not debug into the provided dylib since it has been stripped down.
